### DISCUSSION TITLE
Polyglot piranha for Swift is very slow

### DISCUSSION THREAD

**Discussion by eshi13** [uber/piranha#657] [Q&A]
How long does it take for `execute_piranha` to run on a file of 4000 lines. For me it is running from a very long time. Is it possible I am doing something wrong here. Thanks for help.

**Comment by ketkarameya** [comment]
what language are u using?

**Reply by eshi13** [reply]
swift

**Reply by ketkarameya** [reply]
Swift is expected to be a bit to very slow because of tree-sitter-swift performance issues. :( Are u using Swift?

**Reply by eshi13** [reply]
is there any workaround for that?

**Reply by ketkarameya** [reply]
reach out to me on DM, and maybe i can help you more once i know what your API looks like.  Based on the shape of the API we could probably use concrete-syntax for matching ur API patterns.

**Reply by ketkarameya** [reply]
since concrete syntax circumvents tree-sitter query compilation, things will be significantly faster.  
There is no particular documentation around concrete syntax.

**Reply by ketkarameya** [reply]
@danieltrt  can u please throw in a few sentences around what `CS` is and some examples in the readme?

**Reply by danieltrt** [reply]
Sure. There is one small example here but it's not very elaborate: https://uber.github.io/piranha/docs/reference/dsl/rules

**Reply by ketkarameya** [reply]
@eshi13 note that concrete syntax is very experimental. 
Its accurate at what it does, but at this moment it is not very expressive.

**Reply by ketkarameya** [reply]
some more examples of CS -  https://github.com/uber/piranha/blob/master/src/cleanup_rules/java_cs/rules.toml
